### ISSUE TITLE
terminate called after throwing an instance of 'std::runtime_error'

### ISSUE THREAD

**Issue by osxme** [madMAx43v3r/chia-plotter#11]
I have an E3-1230 V2 4 core, 4 thread CPU, 8G RAM and 512GB SSD.
My command is ./chia_plot xxxxx xxxxx /mnt/512g/caches/ /mnt/512g/files/ 4 128

The following error occurred, what is wrong?

[P1] Table 1 took 366.715 sec
terminate called after throwing an instance of 'std::runtime_error'
  what():  thread failed with: key_shift < 0

**Comment by d3athgu1s3** [comment]
the bucket isn't 128 i believe? it was 2^x , try 7 ?

**Comment by einsz** [comment]
You set [log_num_buckets] to 128. This is not actually the number of buckets, but 2^[log_num_buckets]. So you actually need to set it to 7.

**Comment by VertiHydro** [comment]
Buckets refers to the number to make 2 to the power of. So you should put 7, which means 2^7 buckets, which means 128. Instead, you put 128, which means 2^128 buckets, which means 34028... (34 more digits).

**Comment by Hardcore-fs** [comment]
Also i don't think 512Gb SSD is going to make it.
there are a lot of intermediate files produced, i needed nearly 800 GB free.

**Comment by madMAx43v3r** [comment]
yeah, 2^128 buckets is not gonna work, use 7 instead. chia_plot needs about 300GB tmp space total.

**Comment by madMAx43v3r** [comment]
800GB is for the `test_phase_x` binaries, they don't delete files from previous phases, for testing purposes.

**Comment by Phlexx0** [comment]
Similar issue 
```
./chia_plot XXX XXX  /mnt/d/ /mnt/d/ 4 7
Number of Threads: 4
Number of Sort Buckets: 2^7 (128)
terminate called after throwing an instance of 'std::invalid_argument'
  what():  G1Element::FromBytes: Invalid size
Aborted
```

**Comment by madMAx43v3r** [comment]
one if your keys is too short or too long

**Comment by Phlexx0** [comment]
> one if your keys is too short or too long

Used the wrong set of keys. Didnt realize GI keys are not the same as `./chia.exe show keys` farmer/pool keys

**Comment by osxme** [comment]
Yes, thank you, it should be set to 7 buckets. It worked.
However, my plotting time is very long, [P1] Table 1 took 275 sec, if I follow this time, my final time may exceed 13 hours. This is longer than 7 hours in windows ploting. I wonder if there are several reasons:

1. My cpu has only 4 cores and 4 threads
2. My ssd is only 512gb
3. My tmp disk format is ntfs...

I am using ubuntu 20

**Comment by madMAx43v3r** [comment]
Could be the NTFS indeed...
